shimmy/dm_lab_compatibility.py

```diff
@@ -0,0 +1,81 @@
+"""Wrapper to convert a dm_lab environment into a gymnasium compatible environment."""
+from __future__ import annotations
+
+from typing import Any, TypeVar
+
+import gymnasium
+import numpy as np
+from gymnasium.core import ObsType
+
+from shimmy.utils.dm_lab import dm_lab_obs2gym_obs_space, dm_lab_spec2gym_space
+
+
+class DmLabCompatibility(gymnasium.Env[ObsType, np.ndarray]):
+    """A compatibility wrapper that converts a dm_lab-control environment into a gymnasium environment."""
+
+    metadata = {"render_modes": None, "render_fps": 10}
+
+    def __init__(
+        self,
+        env: Any,
+        render_mode: str | None = None,
+    ):
+        """Initialises the environment with a render mode along with render information."""
+        self._env = env
+
+        # need to do this to figure out what observation spec the user used
+        self._env.reset()
+        self.observation_space = dm_lab_obs2gym_obs_space(self._env.observations())
+        self.action_space = dm_lab_spec2gym_space(env.action_spec())
+
+        assert (
+            render_mode is None
+        ), "Render mode must be set on dm_lab environment init. Pass `renderer='sdl'` to the config of the base env to enable human rendering."
+        self.render_mode = render_mode
+
+    def reset(
+        self, *, seed: int | None = None, options: dict[str, Any] | None = None
+    ) -> tuple[ObsType, dict[str, Any]]:
+        """Resets the dm-lab environment."""
+        super().reset(seed=seed)
+
+        self._env.reset(seed=seed)
+        info = {}
+
+        return (
+            self._env.observations(),
+            info,
+        )  # pyright: ignore[reportGeneralTypeIssues]
+
+    def step(
+        self, action: np.ndarray
```

Comment:
The action type hint is `np.ndarray` however you run `action.values()`, should this be `dict` or can the later line be changed to `actions.astype(np.intc)`